src/routes/dashboard/containers/__tests__/DashboardContainer.spec.js

```diff
@@ -0,0 +1,254 @@
+// @flow
+
+import type {
+  CategoryRouteStateType,
+  CitiesStateType,
+  CityContentResourceCacheStateType,
+  LanguageResourceCacheStateType,
+  StateType
+} from '../../../../modules/app/StateType'
+import CityModelBuilder from '../../../../testing/builder/CityModelBuilder'
+import LanguageModelBuilder from '../../../../testing/builder/LanguageModelBuilder'
+import CategoriesMapModelBuilder from '../../../../testing/builder/CategoriesMapModelBuilder'
+import { flatMap, reduce } from 'lodash'
+import configureMockStore from 'redux-mock-store'
+import React from 'react'
+import { Provider } from 'react-redux'
+import createNavigationScreenPropMock from '../../../../modules/test-utils/createNavigationScreenPropMock'
+import { ScrollView, Text } from 'react-native'
+import TestRenderer from 'react-test-renderer'
+import { render } from '@testing-library/react-native'
+import CategoriesRouteStateView from '../../../../modules/app/CategoriesRouteStateView'
+import brightTheme from '../../../../modules/theme/constants/theme'
+import moment from 'moment'
+import { LanguageModel } from '@integreat-app/integreat-api-client'
+
+jest.mock('react-i18next')
+jest.useFakeTimers()
+
+const mockStore = configureMockStore()
+
+class MockDashboard extends React.Component<{}> {
+  render () {
+    return <Text>Dashboard</Text>
+  }
+}
+
+describe('DashboardContainer', () => {
+  const [city] = new CityModelBuilder(1).build()
+  const languages = new LanguageModelBuilder(2).build()
+  const language = languages[0]
+  const categoriesMap = new CategoriesMapModelBuilder(city.code, language.code).build()
+  const resourceCache: LanguageResourceCacheStateType = {
+    'some-path': {
+      'some-url': {
+        filePath: 'some-file-path',
+        lastUpdate: moment('2016-02-01T10:35:20Z'),
+        hash: '12345678'
+      }
+    }
+  }
+
+  const prepareState = (
+    routeState: ?CategoryRouteStateType,
+    {
+      switchingLanguage,
+      cities,
+      languages,
+      resourceCacheState
+    }: {|
+      switchingLanguage?: boolean,
+      cities?: CitiesStateType,
+      languages?: ?Array<LanguageModel>,
+      resourceCacheState?: CityContentResourceCacheStateType
+    |} = {}
+  ): StateType => {
+    return {
+      darkMode: false,
+      cityContent: {
+        city: city.code,
+        switchingLanguage: switchingLanguage !== undefined ? switchingLanguage : false,
+        languages: languages !== undefined ? languages : [language],
+        categoriesRouteMapping: routeState ? { 'route-id-0': routeState } : {},
+        eventsRouteMapping: {},
+        resourceCache: resourceCacheState || { status: 'ready', value: resourceCache },
+        searchRoute: null
+      },
+      contentLanguage: 'de',
+      cities: cities || { status: 'ready', models: [city] }
+    }
+  }
+
+  const rootCategory = categoriesMap.findCategoryByPath(`/${city.code}/${language.code}`)
```

Comment:
You can also find the root category by finding the category where `category.root == true`

Reply:
Seems more complicated to me.